common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -143,15 +177,30 @@ private void loadUser(UUID uuid, String userName) {
                     RenderSystem.recordRenderCall(() -> Services.Cosmetics.loadCosmeticTextures(uuid, user));
                 },
                 onError -> {
-                    errorCount++;
-                    if (errorCount >= MAX_ERRORS) {
-                        WynntilsMod.error("Athena user lookup has repeating failures. Disabling future lookups.");
+                    errors.put(System.currentTimeMillis());
+
+                    // Save user failures
+                    userFailures.computeIfAbsent(uuid, k -> 0);
+                    userFailures.compute(uuid, (k, v) -> v + 1);
+
+                    // Only log the error once
+                    if (errors.size() == MAX_ERRORS + 1) {
```

Comment:
Can this happen? If errors.size() == MAX_ERRORS we will exit early. (I think the `+1` needs to go)

Reply:
Fixed.